alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/RestIndexMonitorAction.kt

```diff
@@ -180,6 +184,36 @@ class RestIndexMonitorAction(
             }
         }
 
+        /**
+         * This function checks whether the [Monitor] has an [HttpInput] with localhost. If so, make sure the port is same as specified in settings.
+         */
+        private fun validateLocalPort(monitor: Monitor, settingsPort: Int) {
+            if (monitor.inputs.isNotEmpty() && monitor.inputs.single() is HttpInput) {
```

Comment:
I know we currently limit the number of inputs to 1 but no need to limit this code to force only one. Just iterate through all the `input` and any input of type `HttpInput` check for the localhost and port is valid.

Reply:
Fixed. Changed to go through `monitor.inputs` and do validation if it is a `HttpInput`.